cmd/cmdcommon/cmdcommon.go

```diff
@@ -46,6 +49,48 @@ func (h HcCmdHelper) InitiateCommand() {
 	h.checkNameSpace()
 }
 
+const pprofAddrEnvVar = "HCO_PPROF_ADDR"
+
+// Registers a pprof server for cpu and memory profiling the running operator.
+func (h HcCmdHelper) RegisterPPROFServer(mgr manager.Manager) error {
+	pprofAddr := os.Getenv(pprofAddrEnvVar)
+	if len(pprofAddr) == 0 {
+		return nil
+	}
+
+	h.Logger.Info("Registering pprof server.")
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/debug/pprof/", pprof.Index)
+	mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+
+	s := &http.Server{Addr: pprofAddr, Handler: mux}
```

Comment:
I would use the env variable as just a switch like "PPROF_ENABLE=true" since the address must be like `localhost:<port>` in the cluster and users may set it with wrong values with this format. Using a fixed port doesn't seem dangerous to me as well.

Reply:
I don't think normal users would interact much if at all with this flag, so ease of use is not really a concern to me.

Hardcoded ports/addresses is a no go for me, as you can't start the operator locally if there is a port conflict.
Having to reconfigure half of your local stack, just because `:8080` is super popular is really annoying :D

Same with `localhost` in there, if you are in a pod, you probably want to bind to 0.0.0.0/:: while locally you prop want to include localhost.